src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,52 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exceptions.RegistrationException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MIN_AGE = 18;
+    private static final int MIN_CHAR_COUNT = 6;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        if (user.getLogin() == null || user.getLogin().isEmpty()) {
+            throw new RegistrationException("Login can't be null"
+                    + "Min char count is "
+                    + MIN_CHAR_COUNT);
+        }
+        if (user.getPassword() == null || user.getPassword().isEmpty()) {
+            throw new RegistrationException("Password can't be null"
+                    + "Min char count is "
+                    + MIN_CHAR_COUNT);
```

Comment:
Similar to the previous comment, the error message here is not accurate. The message states that the password can't be null and then mentions the minimum character count. However, you're not checking for minimum character count in this if block. You should separate the checks for null or empty password and password length into different if blocks, similar to what you've done in lines 39-44.